frappe/website/context.py

```diff
@@ -62,22 +81,18 @@ def build_context(context):
 				context[prop] = getattr(context.doc, prop, False)
 
 	elif context.controller:
-		module = frappe.get_module(context.controller)
-
-		if module:
-			# get config fields
-			for prop in ("base_template_path", "template", "no_cache", "no_sitemap",
-				"condition_field"):
-				if hasattr(module, prop):
-					context[prop] = getattr(module, prop)
-
-			if hasattr(module, "get_context"):
-				ret = module.get_context(context)
-				if ret:
-					context.update(ret)
-
-			if hasattr(module, "get_children"):
-				context.children = module.get_children(context)
+		# controller based context
+		update_controller_context(context, context.controller)
+		
+		# controller context extensions
+		context_controller_hooks = frappe.get_hooks("extend_controller_context") or {}
+		for controller, extension in context_controller_hooks.items():
```

Comment:
can we call this `extend_website_page_controller_context`, because controller by default means doctype controller